src/autogluon_dashboard/scripts/widget.py

```diff
@@ -4,6 +4,63 @@
 
 
 class Widget:
+    """
+    This class is used to define any Panel widgets that are used on the dashboard website.
+    These widgets can be independent or attached to a dashboard to control the plot.
+    ...
+
+    Attributes
+    ----------
+    widget_type: str
+        the widget type is used to decide what type of panel widget will be created [refer to the if-else statements in __init__]
+    name: str
+        name of the widget on the website
+    value: (int/str)
+        for applicable widgets, this is the value that will be displayed [refer to number and slider widgets]
+    options: (str/List[str])
+        for applicable widgets, this is the list of options that will be displayed [refer to select and toggle widgets]
+    format: str
+        for applicable widgets, this is the format of the "value" being displayed [refer to the number indicator widget]
+    start: int
+        for applicable widgets, this is the start of a range being displayed [refer to the slider widget]
+    end: int
+        for applicable widgets, this is the end of a range being displayed [refer to the slider widget]
```

Comment:
So many of these attributes are only applied to a certain type Widget. What would be matter might be to have a sub-class of widget specifically for this type, instead of every Widget objects having bunch of attributes that are not needed for them.